Raise prefer_ilm configuration as a known issue (#4505)
elastic/observability-docs: docs/en/observability/apm/known-issues.asciidoc

// If applicable, link to fix
////

[discrete]
== `prefer_ilm` required in component templates to create custom lifecycle policies

> _Elastic Stack versions: 8.15.0+_

Reviewer: Isn't it from `8.15.1` onwards?

> _Elastic Stack versions: 8.15.0+_

Author: ```suggestion
_Elastic Stack versions: 8.15.1+_
```

> // The conditions in which this issue occurs
> The issue occurs when creating a _new_ cluster using any 8.15.x version.
> The issue occurs for any APM data streams created from 8.15.x.
> The issue does _not_ occur if custom component template has been created before 8.15.0.

Reviewer 2: Should these be 8.15.1+ too?

```suggestion
// The conditions in which this issue occurs
The issue occurs when creating a _new_ cluster using any 8.15.x version.
The issue occurs for any APM data streams created from 8.15.x.
The issue does _not_ occur if custom component template has been created before 8.15.0.
```

// Describe why it happens

> In 8.15.0, APM Server switched to use https://github.com/elastic/elasticsearch/tree/main/x-pack/plugin/apm-data[apm-data plugin]
> to manage data streams, ingest pipelines, lifecycle policies, etc. In 8.15.1, to fix older clusters using
> default ILM policies and producing unmanaged indices, a fix was added to fallback to default ILM policy if
> it exists. As part of the fix `prefer_ilm` configuration was set to `false`. This setting would have an
> effect if both ILM and DSL were in play - a condition that would occur when trying to configure custom
> ILM policies using `@custom` component templates for the conditions mentioned above.

Reviewer 2: ```suggestion
In 8.15.0, APM Server began using the https://github.com/elastic/elasticsearch/tree/main/x-pack/plugin/apm-data[apm-data plugin] 
to manage data streams, ingest pipelines, lifecycle policies, and more. In 8.15.1, a fix was introduced to address 
unmanaged indices in older clusters using default ILM policies. This fix added a fallback to the default ILM policy 
(if it exists) and set the `prefer_ilm` configuration to `false`. This setting impacts clusters where both ILM and 
data stream lifecycles (DSL) are in effect—such as when configuring custom ILM policies using `@custom` component 
templates, under the conditions mentioned above.
```
